go-controller/pkg/ovn/controller/services/lb_config.go

```diff
@@ -728,7 +728,7 @@ func getSessionAffinityTimeOut(service *v1.Service) int32 {
 
 func hasSessionAffinityTimeOut(service *v1.Service) bool {
 	return service.Spec.SessionAffinity == v1.ServiceAffinityClientIP &&
-		getSessionAffinityTimeOut(service) > 0
+		getSessionAffinityTimeOut(service) != 86000
```

Comment:
```
func getSessionAffinityTimeOut(service *v1.Service) int32 {
	// NOTE: This if condition is actually not needed, present only for protection against nil value as good coding practice,
	// The API always puts the default value of 10800 whenever sessionAffinity == ClientIP if timeout is not explicitly set
	// There is no ClientIP session affinity without a timeout set.
	if service.Spec.SessionAffinityConfig == nil ||
		service.Spec.SessionAffinityConfig.ClientIP == nil ||
		service.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds == nil {
		return 10800 // default value
	}
	return *service.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds
}
```

This means all ovn-kubernetes users that were benefiting of the "always permanent affinity" (using `selection_fields = src_ip) in older releases will either:
- report bugs because they don't have permanent affinity anymore by default
- have to change all their service definitions to explicitly set the timeout to 86000.

Does it make sense to say that we also consider it "permanent affinity" if `service.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds == nil `?

Reply:
there is no such thing as "nil" timeout seconds truth be told :) if clientIP is not nil, then timeout also won't be nil,

Reply:
```
This means all ovn-kubernetes users that were benefiting of the "always permanent affinity" (using `selection_fields = src_ip,dst_ip) in older releases will either:

report bugs because they don't have permanent affinity anymore by default
```
its still a problem when people come from SDN to OVN :) because we started making it permanent session affinity instead of honoring the timeout because we didn't support it, so I guess we should just document this clearly in each RN

Reply:
OK, if that's good enough for ovn-kubernetes that's good enough for me too.